File: Vohra_ProMongoDB_SourceCode/ch3/scripts/saveDocument.php
<?php
 

try 
{
$connection = new MongoClient(); 

$collection=$connection->local->catalog;
$doc1 = array("catalogId" => 'catalog1', "journal" => 'Oracle Magazine', "publisher" => 'Oracle 

Publishing', "edition" => 'November December 2013',"title" => 'Engineering as a 

Service',"author" => 'David A. Kelly');
$status=$collection->insert($doc1);
var_dump($status);
print '<br/>';
$doc2 =array("catalogId" => 'catalog2', "journal" => 'Oracle Magazine', "publisher" => 'Oracle Publishing', "edition" => 'November December 2013',"title" => 'Quintessential and Collaborative',"author" => 'Tom Haunert');
$status=$collection->insert($doc2);
var_dump($status);
print '<br/>';
$doc1['edition']= '11-12-2013';
$doc1['author'] = 'Kelly, David A.';
$doc1['updated']=true;

$status=$collection->save($doc1);
var_dump($status);
print '<br/>';

$doc2['edition']='11-12-2013';
$doc2['author'] = 'Haunert, Tom';
$doc2['updated']=true;

$status=$collection->save($doc2);
var_dump($status);
print '<br/>';

}catch ( MongoConnectionException $e ) 
{
    echo '<p>Couldn\'t connect to mongodb</p>';
    exit();
}catch(MongoCursorException $e) {
 echo '<p>w option is set and the write has failed</p>';
    exit();

}

?>

File: Vohra_ProMongoDB_SourceCode/ch4/mongodbscripts/saveDocument.rb
require 'mongo'
include Mongo

mongo_connection = MongoClient.new 
db=mongo_connection.db("local")
collection=db.create_collection("catalog")

document1={
  "catalogId" => "catalog1",
  "journal" => "Oracle Magazine",
  "publisher" => "Oracle Publishing",
  "edition" =>  "November December 2013",
  "title" => "Engineering as a Service",
  "author" =>  "David A. Kelly"
}

 

print id = collection.insert(document1)
print "\n"
document1={
  "_id" => id,
  "catalogId" => "catalog2",
  "journal" => "Oracle Magazine",
  "publisher" => "Oracle Publishing",
  "edition" =>  "November December 2013",
  "title" => "Quintessential and Collaborative",
  "author" =>  "Tom Haunert"
}


print id = collection.save(document1)
